core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -591,6 +592,67 @@ object Validator {
 
   }
 
+  private def validateInputObject(t: __Type): IO[ValidationError, Unit] = {
+    // https://spec.graphql.org/June2018/#IsInputType()
+    def isInputType(t: __Type): Either[__Type, Unit] = t.kind match {
+      case __TypeKind.LIST | __TypeKind.NON_NULL                         => t.ofType.fold[Either[__Type, Unit]](Left(t))(isInputType)
```

Comment:
`Option[__Type]` would be simpler no?

Reply:
I didn't like it since it's not super clear, by just looking at it, when it's an error or not (having `Some` would be the error state)